engine/src/player-platform.cpp

```diff
@@ -1437,7 +1449,19 @@ Exec_stat MCPlayer::setprop(uint4 parid, Properties p, MCExecPoint &ep, Boolean
 	{
 #ifdef /* MCPlayer::setprop */ LEGACY_EXEC
         case P_FILE_NAME:
-            if (filename == NULL || data != filename)
+        {
+            // Edge case: Suppose filenameA is a valid relative path to defaultFolderA, but invalid relative path to defaultFolderB
+            // 1. Set defaultFolder to defaultFolderB. Set the filename to filenameA. Video will become empty, since the relative path is invalid.
+            // 2. Change the defaultFolder to defaultFolderA. Set the filename again to filenameA. Now the relative path is valid
+            char *t_resolved_filename;
+            bool t_success = false;
+            t_success = resolveplayerfilename(filename, t_resolved_filename);
```

Comment:
I think this needs to be done slightly differently.

The player probably needs to store the resolved (absolute) filename to compare with.

It should resolve here and if the resolved filename is different from the one currently being used, it needs to re-prepare.

You mustn't re-resolve the path on prepare though as things might have changed in between - you need to only resolve once and use that filename. Thus, you'll need to make a variant of prepare that takes the filename to use.

The filename property should still reflect what it was set to, but prepare should be responsible for setting the internal 'absolute' path.